src/com/koxudaxi/ruff/RuffPackageManagerListener.kt

```diff
@@ -9,15 +9,27 @@ import com.jetbrains.python.packaging.PyPackageManager
 class RuffPackageManagerListener(project: Project) : PyPackageManager.Listener {
     private val ruffConfigService = RuffConfigService.getInstance(project)
     private val ruffCacheService = RuffCacheService.getInstance(project)
+    private val ruffMacroService = RuffMacroService.getInstance(project)
     @Suppress("UnstableApiUsage")
     private val lspServerManager = if (lspIsSupported) LspServerManager.getInstance(project) else null
     override fun packagesRefreshed(sdk: Sdk) {
-        ruffConfigService.projectRuffExecutablePath = findRuffExecutableInSDK(sdk, false)?.absolutePath
-        ruffConfigService.projectRuffLspExecutablePath = findRuffExecutableInSDK(sdk, true)?.absolutePath
+        updateMacro(sdk)
+        ruffConfigService.projectRuffExecutablePath = findRuffExecutableInSDK(sdk, false)?.absolutePath?.let {
+            ruffMacroService.collapseProjectExecutablePath(it)
+        }
+        ruffConfigService.projectRuffLspExecutablePath = findRuffExecutableInSDK(sdk, true)?.absolutePath?.let {
+            ruffMacroService.collapseProjectExecutablePath(it)
+        }
         ruffCacheService.setVersion()
         if (lspServerManager != null && ruffConfigService.useRuffLsp) {
             @Suppress("UnstableApiUsage")
             lspServerManager.stopAndRestartIfNeeded(RuffLspServerSupportProvider::class.java)
         }
     }
+
+    private fun updateMacro(sdk: Sdk) {
+        sdk.homeDirectory?.parent?.presentableUrl?.let {
+            ruffMacroService.addMacroExpand(PY_INTERPRETER_DIRECTORY_MACRO_NAME, it)
+        }
+    }
```

Comment:
The timing of the sdk path change is
I want to fire this event there if and when the sdk path changes.